### PR TITLE
Update to Napalm version 3.2.0

### PR DESCRIPTION
**Pull request by nniehoff**
This PR is to update the onboarding plugin to Napalm 3.2.0.  I had to update all dependencies with poetry hence the updates to poetry.lock.

### CODE REVIEW

**Comment by nniehoff** [comment]
I also did some cleanup for NetBox 2.9 (configuration.py) and pylint.

**Comment by dgarros** [comment]
Could we support both napalm 2 and napalm 3 ? if we don't have to enforce a specific version I think it's better. 
poetry by default limit 

Looking at it more closely, I think our pyproject.toml is too restrictive right now, we shouldn't pin our dependencies to a specific version 
```
# from pyproject.toml
invoke = "^1.4.1"
napalm = "^3.2.0"
```
https://python-poetry.org/docs/dependency-specification/

File: pyproject.toml [hunk 1]
```diff
@@ -20,7 +20,8 @@ packages = [
 [tool.poetry.dependencies]
 python = "^3.6 || ^3.7 || ^3.8"
 invoke = "^1.4.1"
```
Comment (glennmatthews) [inline pyproject.toml:22]: Mind moving `invoke` to `dev-dependencies`? AFAIK it's not a requirement for actually using this plugin.

File: pyproject.toml [hunk 2]
```diff
@@ -52,6 +53,42 @@ exclude = '''
...
+# Disabled due to noise: too-few-public-methods, too-many-ancestors, too-many-instance-attributes
+disable = """,
+    line-too-long,
+    bad-continuation,
+    too-few-public-methods,
+    too-many-ancestors,
+    too-many-instance-attributes,
+    duplicate-code,
+    raise-missing-from,
+    super-with-arguments
```
Comment (glennmatthews) [inline pyproject.toml:82]: Do we have an issue open for fixing the `raise-missing-from` and `super-with-arguments` items?

File: tasks.py [hunk 3]
```diff
@@ -44,6 +44,30 @@ def build(context, netbox_ver=NETBOX_VER, python_ver=PYTHON_VER):
...
+        context (obj): Used to run specific commands
+        netbox_ver (str): NetBox version to use to build the container
+        python_ver (str): Will use the Python version docker image to build from
+    """
+    container_name = f"{BUILD_NAME}_netbox_package"
+    context.run(
+        f"docker rm {container_name} || true", env={"NETBOX_VER": netbox_ver, "PYTHON_VER": python_ver}, pty=True,
+    )
+    context.run(
+        f"docker-compose  -f {COMPOSE_FILE} -p {BUILD_NAME} run --name {container_name} -w /source netbox poetry build",
```
Comment (glennmatthews) [inline tasks.py:61]: Is this task needed, and does it really need to use docker? Can we not just run `poetry build` directly?

**Comment by glennmatthews** [comment]
@dgarros If I'm reading that link correctly, `^1.4.1` in pyproject.toml means `>=1.4.1, <2.0.0`, which seems reasonable to me.

Agreed that it would be nice (if possible) to set the `napalm` dependency to something like `>=2.5.0, <4.0.0` instead of requiring at least 3.2.0.

**Comment by nniehoff** [comment]
> Could we support both napalm 2 and napalm 3 ? if we don't have to enforce a specific version I think it's better.
> poetry by default limit
> 
> Looking at it more closely, I think our pyproject.toml is too restrictive right now, we shouldn't pin our dependencies to a specific version
> 
> ```
> # from pyproject.toml
> invoke = "^1.4.1"
> napalm = "^3.2.0"
> ```
> 
> https://python-poetry.org/docs/dependency-specification/

AIUI right now we are basically saying for napalm we want versions >= 3.2.0 < 4.0.0.  We could change this to something like:

```
napalm = ">= 2.5.0, < 4"
```

I think the problem is if version 4 comes out I don't think we want to automatically support that without testing so at the very least I think we need to keep the < 4 just to be safe.  If we want to add support for multiple versions we will also need to adjust the tests for multiple versions.

File: tasks.py (hunk 3 above)
Comment (nniehoff) [reply tasks.py:61]: perhaps not, I just wanted it to align with our other plugins and it's a handy method to have.  The nice part about doing this in docker is it doesn't require me to have poetry installed in my development environment.

File: pyproject.toml (hunk 2 above)
Comment (nniehoff) [reply pyproject.toml:82]: https://github.com/networktocode/ntc-netbox-plugin-onboarding/issues/115

File: tasks.py (hunk 3 above)
Comment (mzbroch) [reply tasks.py:61]: +1 here for `poetry build` directly

**Comment by mzbroch** [comment]
> Could we support both napalm 2 and napalm 3 ? if we don't have to enforce a specific version I think it's better.
> poetry by default limit

I agree, in general we're more focused on NAPALM facts structure and its content, than a specific NAPALM version.

File: pyproject.toml (hunk 1 above)
Comment (nniehoff) [reply pyproject.toml]: done

**Review by mzbroch — CHANGES_REQUESTED** [review]
LGTM, please squash commits

**Review by mzbroch — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**